searching/exponential_search.c

```diff
@@ -0,0 +1,99 @@
+/**
+ *  \file
+ *  \@brief [Exponential Search](https://github.com/TheAlgorithms/Algorithms-Explanation/blob/master/en/Search%20Algorithms/Exponential%20Search.md)
+ *  \author [Alessio Farinelli](https://github.com/faridevnz)
+ */
+#include <assert.h>  /// for assert
+
+#define ELEMENT -10
+
+int binary_search(const int* arr, const int l_index, const int r_index, const int n);
+int exponential_search(const int* arr, const int length, const int n);
+static void test();
+
+/**
+ *  Function: exponential_search
+ *  ----------------------------
+ *  \param arr:         array where search the element
+ *  \param length:      the total length of the given array (arr)
+ *  \param n:           element to find in the array (arr)
+ * 
+ *  \returns:           the index of the element (n) in the array (arr)
+ *  \returns:           -1 if the element wasn't found
+ */
+int exponential_search(const int* arr, const int length, const int n) 
+{
+    // find the upperbound
+    int upper_bound = 1;
+    while ( upper_bound <= length && arr[upper_bound] < n ) { upper_bound = upper_bound * 2; }
+    // calculate the range ( between lower_boud and upper_bound )
+    int lower_bound = upper_bound/2;
+    if ( upper_bound > length ) { upper_bound = length; }
+    // apply the binary search in the range
+    return binary_search(arr, lower_bound, upper_bound, n);
+}
+
+/**
+ *  Function: binary_search
```

Comment:
Provide a brief description of what the function does or is for. For long/detailed description, use `\details`.